dashboard/test/controllers/script_levels_controller_test.rb

```diff
@@ -65,6 +65,36 @@ class ScriptLevelsControllerTest < ActionController::TestCase
     AzureTextToSpeech.unstub(:get_voices)
   end
 
+  test 'should show script level for csp1-2020 lockable lesson with lesson plan' do
+    @unit = create :script, name: 'csp1-2020'
+    @lesson_group = create :lesson_group, script: @unit
+    @lockable_lesson = create(:lesson, script: @unit, name: 'Assessment Day', lockable: true, lesson_group: @lesson_group, has_lesson_plan: false, absolute_position: 15, relative_position: 2)
+    @level_group = create(:level_group, :with_sublevels, name: 'assessment 1')
+    @lockable_level_group_sl = create(:script_level, script: @unit, lesson: @lockable_lesson, levels: [@level_group], assessment: true)
+    get :show, params: {
+      script_id: @unit.name,
+      stage_position: @lockable_lesson.absolute_position - 1,
+      id: @lockable_level_group_sl.position
+    }
+
+    assert_redirected_to '/s/csp1-2020/lockable/2/puzzle/1/page/1'
+  end
+
+  test 'should show script level for csp2-2020 lockable lesson with lesson plan' do
+    @unit = create :script, name: 'csp2-2020'
+    @lesson_group = create :lesson_group, script: @unit
+    @lockable_lesson = create(:lesson, script: @unit, name: 'Assessment Day', lockable: true, lesson_group: @lesson_group, has_lesson_plan: false, absolute_position: 9, relative_position: 1)
+    @level_group = create(:level_group, :with_sublevels, name: 'assessment 1')
+    @lockable_level_group_sl = create(:script_level, script: @unit, lesson: @lockable_lesson, levels: [@level_group], assessment: true)
+    get :show, params: {
+      script_id: @unit.name,
+      stage_position: @lockable_lesson.absolute_position,
+      id: @lockable_level_group_sl.position
+    }
+
+    assert_redirected_to '/s/csp2-2020/lockable/1/puzzle/1/page/1'
```

Comment:
ahh, I had not thought about the subpage urls. these urls will need to also work at their new locations before the step of "make new urls start working" can be considered to be complete. can you please add tests showing that those urls work? I think the way you've written the above unit tests looks good enough given what we're testing and how long we'll need them for.

Reply:
I'm not sure what you are looking for that is different from what is here. Can you clarify?

Reply:
please correct me if I am mistaken but it looks like you are testing what happens when you go to /s/csp2-2020/stage/15/puzzle/1/ . I am asking what happens at /s/csp2-2020/stage/15/puzzle/1/page/1 .

Reply:
Ah I see what you are saying. Going directly to that URL instead of the redirect. I'll add something